packages/material-ui/src/InputBase/InputBase.js

```diff
@@ -482,6 +482,7 @@ const InputBase = React.forwardRef(function InputBase(inProps, ref) {
   const Input = components.Input || InputBaseComponent;
   inputProps = { ...inputProps, ...componentsProps.input };
 
+  delete other.hiddenLabel; // prevent hiddenLabel leak to DOM
```

Comment:
I think this is the direct way to prevent leaking, I tried with another approach but it makes the code a lot more complicated.

Reply:
Note `hiddenLabel` is used for styling only so delete it is fine. It is only needed in `styleProps`.

Reply:
No strong opinion, but for consistency with what we have so far in the other component, I would just destruct it from props. I haven't seen so far this pattern for removing keys from `other`.

Reply:
Fixing the leak at the InputBase looks correct since it's read by `formControlState`. However, we usually restructure the props. Did you consider?

```diff
diff --git a/packages/material-ui/src/InputBase/InputBase.js b/packages/material-ui/src/InputBase/InputBase.js
index 7ea570e61c..261aea49a7 100644
--- a/packages/material-ui/src/InputBase/InputBase.js
+++ b/packages/material-ui/src/InputBase/InputBase.js
@@ -233,6 +233,7 @@ const InputBase = React.forwardRef(function InputBase(inProps, ref) {
     endAdornment,
     error,
     fullWidth = false,
+    hiddenLabel,
     id,
     inputComponent = 'input',
     inputProps: inputPropsProp = {},
```

Reply:
By destructuring we have a single place where we can look where props go. Using destructuring and `delete` "spreads" these locations out over the implementation making it harder to reason about the component. So yeah, this should be moved to destrcturing. Especially since you already recognoized that this is "weird" code and needs a comment.